packages/babel-plugin-transform-typescript/src/index.js

```diff
@@ -39,16 +34,15 @@ export default declare((api, { jsxPragma = "React" }) => {
       Identifier: visitPattern,
       RestElement: visitPattern,
 
-      Program(path, state: State) {
-        state.programPath = path;
-
+      Program(path, state) {
         const { file } = state;
+        let fileJsxPragma = null;
 
         if (file.ast.comments) {
           for (const comment of (file.ast.comments: Array<Object>)) {
             const jsxMatches = JSX_ANNOTATION_REGEX.exec(comment.value);
             if (jsxMatches) {
-              file.set(PRAGMA_KEY, jsxMatches[1]);
+              fileJsxPragma = jsxMatches[1];
```

Comment:
Instead of storing this in a shared mutable state under some artificial key I think it's better to just pass it as explicit argument to `isImportTypeOnly`, as it makes the data flow clearer - especially that this info doesn't have to be passed through layers of functions, it's needed just one 1 level lower.